src/main/kotlin/com/fortysevendeg/hood/Comparator.kt

```diff
@@ -1,33 +1,45 @@
 package com.fortysevendeg.hood
 
+import arrow.core.Option
+import arrow.core.getOrElse
+import arrow.core.toOption
+import arrow.data.extensions.list.foldable.exists
+import arrow.data.extensions.list.foldable.forAll
+import arrow.data.extensions.list.semigroup.plus
+import arrow.data.foldLeft
 import arrow.effects.IO
+import arrow.effects.extensions.io.applicativeError.handleError
+import arrow.effects.extensions.io.fx.fx
 import arrow.effects.fix
-import arrow.effects.instances.io.applicativeError.handleError
-import arrow.effects.instances.io.monad.monad
-import arrow.instances.list.foldable.exists
-import arrow.instances.list.foldable.forAll
+import com.fortysevendeg.hood.models.*
+import com.fortysevendeg.hood.reader.CsvBenchmarkReader
+import com.fortysevendeg.hood.reader.JsonBenchmarkReader
 import java.io.File
 
 object Comparator {
 
-  private fun compare(previous: Benchmark, current: Benchmark, threshold: Int): BenchmarkResult =
+  private fun compare(
+    previous: Benchmark,
+    current: Benchmark,
+    threshold: Double
+  ): BenchmarkResult =
     when {
-      previous.score <= current.score             -> BenchmarkResult.OK
-      previous.score - current.score <= threshold -> BenchmarkResult.WARN
-      else                                        -> BenchmarkResult.FAILED
+      previous.getScore() <= current.getScore() -> BenchmarkResult.OK
+      previous.getScore() - current.getScore() <= threshold -> BenchmarkResult.WARN
+      else -> BenchmarkResult.FAILED
     }
 
   private fun getCompareResults(
     currentBenchmarks: Map<String, List<Benchmark>>,
     prev: Benchmark,
-    threshold: Int
+    threshold: Double
   ): List<Pair<List<Benchmark>, BenchmarkResult>> =
     currentBenchmarks.mapValues {
-      it.value.filter { current -> prev.key == current.key }
-    }.flatMap {
+      it.value.filter { current -> prev.getKey() == current.getKey() }
+    }.flatMap { entry ->
       val currentWithName: List<Benchmark> =
-        it.value.map { current -> Benchmark(it.key, current.score) }
-      it.value.map { current -> Pair(currentWithName, compare(prev, current, threshold)) }
+        entry.value.map { it.withName(entry.key) }
+      entry.value.map { Pair(currentWithName, compare(prev, it, threshold)) }
```

Comment:
Would this work?
```suggestion
      entry.value.map { Pair(it.withName(entry.key), compare(prev, it, threshold)) }
```

Reply:
It will not because we are changing the benchmarks key in order to distinguish them later and putting them along the result of their comparison